lib/helpers/util.js

```diff
@@ -1,18 +1,50 @@
-module.exports = {
-  toFixed: (value, dp) => {
+const xml2js = require('xml2js')
+const csv = require('csvtojson')
+
+class Util {
+  toFixed (value, dp) {
     if (value) {
       return Number(Math.round(value + 'e' + dp) + 'e-' + dp).toFixed(dp)
     } else {
       return value
     }
-  },
-  isNumeric: n => {
+  }
+
+  isNumeric (n) {
     return !isNaN(parseFloat(n)) && isFinite(n)
-  },
-  parseFloatNull: val => {
+  }
+
+  parseFloatNull (val) {
     return isNaN(parseFloat(val)) ? null : parseFloat(val)
-  },
-  parseIntNull: (val, radix) => {
+  }
+
+  parseIntNull (val, radix) {
     return isNaN(parseInt(val, radix)) ? null : parseInt(val, radix)
   }
+
+  parseXml (body) {
+    return new Promise((resolve, reject) => {
+      xml2js.parseString(body, (err, value) => {
+        if (err) {
+          reject(err)
```

Comment:
If this errors then it will correctly reject but then it will go on to resolve afterwards.
Put a `return` after the `reject(err)` on Line 29.